cmake/aikidoConfig.cmake.in

```diff
@@ -0,0 +1,138 @@
+set(aikido_VERSION x.y.z)
+
+@PACKAGE_INIT@
+
+function(aikido_traverse_components output_variable)
+  set(output_components)
+
+  foreach(component ${ARGN})
+    # We should be able to elminate the <C>_VISITED property by detecting
+    # whether the <C>_FOUND property is DEFINED. Unfortunately, DEFINED checks
+    # whether or not define_property() was called, not whether the variable was
+    # set. The exact behavior seems to vary between CMake versions, so we use
+    # an extra variable instead.
+    get_property(is_visited DIRECTORY PROPERTY "aikido_${component}_VISITED")
+    set_property(DIRECTORY PROPERTY "aikido_${component}_VISITED" TRUE)
+
+    if(NOT is_visited)
+      set(component_include
+        "${CMAKE_CURRENT_LIST_DIR}/aikido_${component}Component.cmake")
+      set(target_include
+        "${CMAKE_CURRENT_LIST_DIR}/aikido_${component}Targets.cmake")
+
+      if(EXISTS "${component_include}" AND EXISTS "${target_include}")
+        include("${component_include}")
+
+        set_property(DIRECTORY PROPERTY "aikido_${component}_FOUND" TRUE)
+        set_property(DIRECTORY
+          PROPERTY "aikido_${component}_INCLUDE_DIRS"
+          ${aikido_${component}_INCLUDE_DIRS})
+        set_property(DIRECTORY
+          PROPERTY "aikido_${component}_DEFINITIONS"
+          ${aikido_${component}_DEFINITIONS})
+        set_property(DIRECTORY
+          PROPERTY "aikido_${component}_LIBRARIES"
+          ${aikido_${component}_LIBRARIES})
+        set_property(DIRECTORY
+          PROPERTY "aikido_${component}_DEPENDENCIES"
+          ${aikido_${component}_DEPENDENCIES})
+
+        aikido_traverse_components("${output_variable}" 
+          ${aikido_${component}_DEPENDENCIES})
+
+        include("${target_include}")
+
+        # This is not necessary in recent versions of cmake, which supports
+        # transitive includes via INTERFACE_INCLUDE_DIRECTORIES.
+        foreach(dependency ${aikido_${component}_DEPENDENCIES})
+          set_property(DIRECTORY APPEND
+            PROPERTY "aikido_${component}_INCLUDE_DIRS"
+            ${aikido_${dependency}_INCLUDE_DIRS})
+          set_property(DIRECTORY APPEND
+            PROPERTY "aikido_${component}_DEFINITIONS"
+            ${aikido_${dependency}_DEFINITIONS})
+          # LIBRARIES is automatically inherited via INTERFACE_LINK_LIBRARIES.
```

Comment:
I think this would be pretty great.  It would eliminate this loop entirely and be forward compliant with transitioning to just using the builtin transitive dependency system in later CMake versions.